### PR TITLE
Added Median class

### LINKED ISSUE #46: Add program to find mean, median, mode of a array

### PR DESCRIPTION
**Pull request by KostasThan**
For #46 added **<ins>ONLY</ins>** the Median class whilist working on the others.

### **Describe your change:**

- [ ] Add an Algorithm?
- [ ] Add a Data Structure?
- [x] Add a Program not related to above?
- [ ] Fix a bug or typo in an existing cases?
- [ ] Documentation change?

### **Checklist:**

- [x] This pull request is all my own work -- I have not plagiarized.
- [x]  I know that pull requests will not be merged if they fail the automated tests.
- [x]  All filenames are in all uppercase characters with no spaces or dashes.
- [x]  All functions and variable names follow Java naming conventions.
- [x]  If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: NumberSystem/Median.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		}
+		//else return the middle one
+		return array[(array.length-1)/2];
+	}
+	
+	
+	
+	/**
+	 * This method returns the median value of an array.
+	 * Note that the returned value is always boolean despite the int[] input.
```
Comment (AkMo3) [inline NumberSystem/Median.java]: I think you meant the returned value is always double.

File: NumberSystem/Median.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                .toArray();
+		System.out.println(Arrays.toString(doubleValues));
+		System.out.println(Median.getMedian(doubleValues));	
+	}
+	
+	
+	
+	/**
+	 * This method returns the median value of an array.
+	 * Note that the returned value is always boolean.
```
Comment (AkMo3) [inline NumberSystem/Median.java]: Returned value is double i suppose

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Run these commands to pass all checks. 

  1.  wget https://github.com/google/google-java-format/releases/download/google-java-format-1.9/google-java-format-1.9-all-deps.jar -O formatter.jar
  2.  java -jar formatter.jar --replace {FILE_NAME.java}

Also follow the commit instructions as far as possible.

Also, you can these changes after your exams. Good luck for them.

File: NumberSystem/Median.java (hunk 1 above)
Comment (KostasThan) [reply NumberSystem/Median.java]: > Run these commands to pass all checks. 

I messed up with my VM so i don't have linux at the time. Is there a way to run them at windows?

>I think you meant the returned value is always double.

Good catch. Thanks!
I am new at this this is my second PR so I do make some newbie mistakes.
Thank you for being kind and understanding!

**Review by AkMo3 — APPROVED** [review]

File: NumberSystem/Median.java (hunk 1 above)
Comment (AkMo3) [reply NumberSystem/Median.java]: > I messed up with my VM so i don't have linux at the time. Is there a way to run them at windows?

Yup you can run them in powershell. These commands works well there.

**Comment by AkMo3** [comment]
Just waiting for formatting to be fixed. Once fixed I would merge.

**Comment by KostasThan** [comment]
I think i managed to do it. Althought there is still an error.

**Comment by AkMo3** [comment]
Looks good to be. That test is giving error due to more than 100 character in single line, but I would take care of that.

**Comment by KostasThan** [comment]
> Looks good to be. That test is giving error due to more than 100 character in single line, but I would take care of that.

Great. Thanks!

**Pull request merged**